### PR TITLE
Package of user experience improvements and docs

### PR DESCRIPTION
**Pull request by alivinco**
### Added
* Added workflow documentation.
* Added `wait_for_event` task. This task will wait for a specific event to occur.Can be used to pause/resume workflow execution , for instance when a user needs to approve the workflow execution.
* Added metrics helper functions. These functions can be used to create metrics for the workflow.
* Added UI configuration editor. Now it supports all UI operations.
* Added workflow source code viewer.
* Added rules file download link. Now user can upload and download rules file via NEAT UI .
* Added error reporting in UI if the rules file is not valid or not present. The same for data exploration view.

### Improved
* Many UI improvements and visual regrouping of UI views.
* Improved http trigger. Now it can receive arbitrary data in json format.
* Global configurations moved to its own view.
* Steps and System components editor supports node removal.

### Changed
* Groups section was renamed to Solution/System components overview. In manifest it was renamed to `system_components`.

### CODE REVIEW

File: cognite/neat/core/data_stores/metrics.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+from prometheus_client import REGISTRY, Counter, Gauge, Metric
+
+
+class NeatMetricsCollector:
+    def __init__(self, name: str, cdf_client: CogniteClient = None) -> None:
+        self.name = name
+        self.metrics: dict[str, Gauge | Counter] = {}
+
+    def register_metric(
+        self, metric_name: str, metric_description: str = "", m_type: str = "gauge", metric_labels: list[str] = None
```
Comment (nikokaoja) [inline cognite/neat/core/data_stores/metrics.py:13]: Suggestion: rename `m_type` to `metric_type`

File: pyproject.toml [hunk 2]
```diff
@@ -16,6 +16,9 @@ exclude = [
     "cognite/neat/explorer-ui/neat-app/node_modules/**",
     "cognite/neat/explorer-ui/neat-app/*.json",
 ]
+[tool.poetry.group.dev.dependencies]
+mkdocs-glightbox = "^0.3.4"
```
Comment (doctrino) [inline pyproject.toml:20]: Needs to be part of the docs dependencies

File: cognite/neat/migration/wf_manifests.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        wf_module_full_path = wf_store_path / wf_module_name
+
+        if wf_module_full_path.is_dir():
+            metadata_file = wf_store_path / wf_module_name / "workflow.yaml"
+            metadata_file_migrated = wf_store_path / wf_module_name / "workflow.yaml"
+            logging.info(f"Loading workflow {wf_module_name} metadata from {metadata_file}")
+            if metadata_file.exists():
+                with open(metadata_file, "r") as f:
+                    # metadata_str = f.read()
+                    manifest_yaml = yaml.load(f, Loader=yaml.Loader)
```
Comment (doctrino) [inline cognite/neat/migration/wf_manifests.py:22]:
```suggestion
                    manifest_yaml = yaml.safe_load(f, Loader=yaml.Loader)
```

File: cognite/neat/explorer/explorer.py [hunk 4]
```diff
@@ -100,7 +103,11 @@ def read_root():
 
 @app.get("/api/about")
 def get_about():
-    return {"version": neat.__version__}
+    response = {"version": neat.__version__}
+    installed_packages = pkg_resources.working_set
+    installed_packages_list = sorted(["%s==%s" % (i.key, i.version) for i in installed_packages])
```
Comment (doctrino) [inline cognite/neat/explorer/explorer.py:108]:
```suggestion
    installed_packages_list = sorted([f"{i.key}=={i.version}" for i in installed_packages])
```

File: cognite/neat/core/data_stores/metrics.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        metric_name = f"neat_workflow_{self.name}_{metric_name}"
+        if metric_name in REGISTRY._names_to_collectors:
+            self.metrics[metric_name] = REGISTRY._names_to_collectors[metric_name]
+        else:
+            if m_type == "gauge":
+                metric = Gauge(metric_name, metric_description, metric_labels)
+            elif m_type == "counter":
+                metric = Counter(metric_name, metric_description, metric_labels)
+            if metric:
+                self.metrics[metric_name] = metric
```
Comment (doctrino) [inline cognite/neat/core/data_stores/metrics.py:27]: I found this slightly hard to read, so just a 🏌 suggestion to make it more readable.
```suggestion
        metric_labels = [] if metric_labels is None else metric_labels

        metric_name = f"neat_workflow_{self.name}_{metric_name}"
        if metric_name in REGISTRY._names_to_collectors:
            self.metrics[metric_name] = REGISTRY._names_to_collectors[metric_name]
            return

        metric = None
        if m_type == "gauge":
            metric = Gauge(metric_name, metric_description, metric_labels)
        elif m_type == "counter":
            metric = Counter(metric_name, metric_description, metric_labels)

        if metric:
            self.metrics[metric_name] = metric
```

**Review by doctrino — APPROVED** [review]
I do not have time to read through the docs. Nothing critical on my part, so I am pre-approving. Suggest going quickly through the comments before merging.

File: cognite/neat/explorer/explorer.py [hunk 6]
```diff
@@ -91,6 +93,7 @@ async def lifespan(app_ref: FastAPI):
 prom_app = make_asgi_app()
 app.mount("/metrics", prom_app)
 app.mount("/static", StaticFiles(directory=constants.UI_PATH), name="static")
+app.mount("/data", StaticFiles(directory=config.data_store_path), name="data")
```
[the remark this answers is not in the document]
Comment (alivinco) [reply cognite/neat/explorer/explorer.py:96]: Main motivation is to allow user to download rule files (download ->modify->upload) or python files

File: cognite/neat/core/data_stores/metrics.py (hunk 1 above)
Comment (alivinco) [reply cognite/neat/core/data_stores/metrics.py:13]: Good point , will update

File: pyproject.toml (hunk 2 above)
Comment (alivinco) [reply pyproject.toml:20]: Me need either this plugin or similar that allows zoom/enlarge images

**Pull request merged**